### ISSUE TITLE
Using AdaDelta causes resuming training to fail with incorrect length of history blobs error 

### ISSUE THREAD

**Issue by Coderx7** [BVLC/caffe#4160]
I tried to resume my pretrained model and found out whenever I try to use a solver which uses AdaDelta, it crashes caffe. other types of learning rate types such as inv,SGD,RMSProp, etc are fine. 
Here is a sample error I get when trying AdaDelta:
```
I0517 09:05:26.691237  5888 solver.cpp:60] Solver scaffolding done.
I0517 09:05:26.694239  5888 caffe.cpp:210] Resuming from examples/mnist/lenet_iter_64000.solverstate
F0517 09:05:26.901387  5888 sgd_solver.cpp:316] Check failed: state.history_size() == history_.size() (57 vs. 114) Incorrect length of history blobs.
*** Check failure stack trace: ***
```

**Comment by naolenikhil** [comment]
I am facing the same problem. Did you find a solution to it?

**Comment by Aniruddh1** [comment]
try second last checkpoint. it worked for me.
